internal/aws/awsutil/conn.go

```diff
@@ -89,8 +89,13 @@ func getProxyAddress(proxyAddress string) string {
 	case proxyAddress != "":
 		finalProxyAddress = proxyAddress
 
-	case proxyAddress == "" && os.Getenv("HTTPS_PROXY") != "":
-		finalProxyAddress = os.Getenv("HTTPS_PROXY")
+	case proxyAddress == "":
+		if os.Getenv("HTTPS_PROXY") != "" {
+			finalProxyAddress = os.Getenv("HTTPS_PROXY")
+		}
+		if os.Getenv("NO_PROXY") != "" {
```

Comment:
this seems incorrect. See https://pkg.go.dev/golang.org/x/net/http/httpproxy#Config 

NO_PROXY stands for a list of comma delimited domain names where proxy should not be applied. It doesn't override HTTPS_PROXY in general, only for some specific domains.

Reply:
yes, I I misunderstand, thx